packages/core/ui-settings/core-ui-settings-browser-internal/src/settings_service.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Subject } from 'rxjs';
+
+import type { InternalInjectedMetadataSetup } from '@kbn/core-injected-metadata-browser-internal';
+import type { HttpSetup } from '@kbn/core-http-browser';
+import type { SettingsStart } from '@kbn/core-ui-settings-browser';
+import { UiSettingsApi } from './ui_settings_api';
+import { UiSettingsClient } from './ui_settings_client';
+import { UiSettingsGlobalClient } from './ui_settings_global_client';
+
+export interface SettingsServiceDeps {
+  http: HttpSetup;
+  injectedMetadata: InternalInjectedMetadataSetup;
+}
+
+/** @internal */
+export class SettingsService {
+  private uiSettingsApi?: UiSettingsApi;
+  private uiSettingsClient?: UiSettingsClient;
+  private uiSettingsGlobalClient?: UiSettingsGlobalClient;
+  private done$ = new Subject();
+
+  public setup({ http, injectedMetadata }: SettingsServiceDeps): SettingsStart {
```

Comment:
Sorry if I wasn't unclear in my previous comment: I'd prefer to still keep a `type SettingsStart = SettingsSetup` (or the other way, but that's usually the way we've been doing it) and have `setup` and `start` return `SettingsSetup` and `SettingsStart` respectively.

Reply:
gotcha, makes sense :)